recipes/parallelproj/meta.yaml

```diff
@@ -0,0 +1,44 @@
+{% set name = "parallelproj" %}
+{% set version = "1.2.6" %}
+{% set build_number = "0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/gschramm/parallelproj/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 826a83e51db6459004d17c030c21fb32e7cb5205a45c28c8f50f87f235321196
+
+build:
+  number: {{ build_number }}
+  string: "h{{ PKG_HASH }}_{{ build_number }}_cpu"   # [cuda_compiler_version == "None"]
+  string: "h{{ PKG_HASH }}_{{ build_number }}_cuda"  # [cuda_compiler_version != "None"]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}  # [cuda_compiler_version != "None"]
+    - cmake >=3.23
+    - ninja
+  host:
+    - libgomp      # [linux]
+    - llvm-openmp  # [osx]
+test:
+  commands:
+    - test -f $PREFIX/lib/libparallelproj_c$SHLIB_EXT  # [unix]
+    - if not exist %LIBRARY_LIB%\parallelproj_c.lib exit 1  # [win]
+    - test -f $PREFIX/lib/libparallelproj_cuda$SHLIB_EXT  # [linux and cuda_compiler_version != "None"]
+    - parallelproj_nontof_fwd_back_test  # [not win]
+    - parallelproj_nontof_fwd_back_test.exe  # [win]
```

Comment:
Do end users need this test binary?

Reply:
No, end users don't need those test binaries. I use those binaries as "ctest" to check whether certain functions of the lib behave as expected. Is there a better way to include e.g. "ctest -VV" in the test section?
I have "ctest -VV" in the build scripts in the beginning, but not sure if this really belongs there.

Reply:
You can run test binaries during the build phase without installing them. You can run test binaries during the test phase without installing them by listing them as test source files.

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-files

Reply:
I added the ```source_file``` section, but the tests fail because the binaries are not found. Do you know why?

Reply:
Source files really have to be in the source, not created during the build phase

Reply:
And is there no way of getting the compiled test executables from the build phase?

Reply:
Yeah you can test during the build phase. Everything that makes it to the test phase will also be installed for endusers

Reply:
```ctest -VV``` is now run during the build phase and the test executables are not installed anymore.

Reply:
Folder structure is preserved, so you would have to call the tests this way:

```
  source_files:
    - build/c/tests/parallelproj_nontof_fwd_back_test  # [not win]
  commands:
    - build/c/tests/parallelproj_nontof_fwd_back_test  # [not win]
```
Note in the logs that the error was "parallelproj_nontof_fwd_back_test not found" during the test phase. Not that "build/c/tests/parallelproj_nontof_fwd_back_test not found" in the source.

Reply:
> Folder structure is preserved, so you would have to call the tests this way:
> 
> ```
>   source_files:
>     - build/c/tests/parallelproj_nontof_fwd_back_test  # [not win]
>   commands:
>     - build/c/tests/parallelproj_nontof_fwd_back_test  # [not win]
> ```
> 
> Note in the logs that the error was "parallelproj_nontof_fwd_back_test not found" during the test phase. Not that "build/c/tests/parallelproj_nontof_fwd_back_test not found" in the source.

Ah ok, sorry for having missed that. I corrected that, but now I get the error
```
build/c/tests/parallelproj_nontof_fwd_back_test: error while loading shared libraries: libparallelproj_c.so.1.2.8: cannot open shared object file: No such file or directory
```

But the lib got installed during the build phase:
```
-- Installing: $PREFIX/lib/libparallelproj_c.so.1.2.8
```

What am I missing?

Reply:
I think sometimes test binaries aren't relocatable, so when you move them out of the same folder as the libraries they link to, they break. I think there's a way to ask conda-build to fix the breaks, but I don't know what it is without looking at the docs.